spec/index.bs

```diff
@@ -2148,21 +2148,83 @@ premultiplication) in which the WebGPU numeric values are to be interpreted.
+<span id=out-of-gamut-premultiplied-rgba-value><!-- historical permalink --></span>
+</h4>
+
+An [=out-of-range premultiplied RGBA value=] is one for which the unpremultiplied representation
+of the same color would be outside of the range of the format.
+
+<div class=example>
+    For example, the unpremultiplied sRGB RGBA value `[1.04, 0, 0, 0.5]` and the
+    premultiplied sRGB RGBA value `[0.52, 0, 0, 0.5]` both represent the
+    CSS color <code><a funcdef>color</a>(srgb 1.04 0 0 / 0.5)</code>, which is equivalent to the
+    in-gamut Display P3 color <code><a funcdef>color</a>(display-p3 0.95 0.21 0.15 / 0.5)</code>.
+
+    When represented in a {{PredefinedColorSpace/"srgb"}} {{GPUTextureFormat/"rgba8unorm"}} canvas,
+    both values are out-of-range for the purposes of this definition,
+    even though the premultiplied value is representable.
+</div>
+
+When a canvas containing such a value is
+[$get a copy of the image contents of a context|used as an image source$],
+the returned image must represent the original color values without clamping,
+regardless of the canvas's {{GPUCanvasConfiguration/format}}.
+
+However, when a ({{GPUCanvasAlphaMode/"premultiplied"}}) canvas containing such a value is
+displayed *on screen*, the visible result for the entire canvas is undefined.
+
+<div class=example>
+    For example, on a display with a color space of exactly Display P3,
+    an on-screen, solid-color {{PredefinedColorSpace/"srgb"}} canvas displays as follows:
+
+    <table class=data>
+        <thead>
+            <tr><th>{{GPUCanvasConfiguration/alphaMode}},<br>{{GPUCanvasConfiguration/format}}<th>Pixel Value<th>Result
+        <tbody>
+            <tr><td>{{GPUCanvasAlphaMode/"opaque"}},<br>{{GPUTextureFormat/"rgba16float"}}
+                <td rowspan=2>`[1.04, 0, 0, 1.0]`
+                <td rowspan=2><code><a funcdef>color</a>(display-p3 0.95 0.21 0.15 / 1.0)</code>.
+            <tr><td>{{GPUCanvasAlphaMode/"premultiplied"}},<br>{{GPUTextureFormat/"rgba16float"}}
+            <tr><td>{{GPUCanvasAlphaMode/"premultiplied"}},<br>{{GPUTextureFormat/"rgba16float"}}
+                <td>`[0.52, 0, 0, 0.5]`
+                <td><code><a funcdef>color</a>(display-p3 0.95 0.21 0.15 / 0.5)</code>.
+            <tr><td>{{GPUCanvasAlphaMode/"premultiplied"}},<br>{{GPUTextureFormat/"rgba8unorm"}}
+                <td>`[0.50, 0, 0, 0.5]`
+                <td><code><a funcdef>color</a>(display-p3 0.92 0.20 0.14 / 0.5)</code>.
+            <tr><td>{{GPUCanvasAlphaMode/"premultiplied"}},<br>{{GPUTextureFormat/"rgba8unorm"}}
+                <td>`[0.52, 0, 0, 0.5]`
+                <td>(undefined result),
+                    <br>because the unpremultiplied color cannot be represented in this format.
+            <!-- POSTV1(#4108): add example for XR formats, which *probably* handle this -->
+    </table>
+
+    Visible results may be undefined because the value could be lost at different steps of display:
```

Comment:
I wanted to write out an example of the actual blending math, but that ends up demonstrating a completely *different* problem, which is that you get very different results doing nonlinear blending in display color space, vs in sRGB.

Reply:
Yes, it gets complicated.